meerk40t/gui/simulation.py

```diff
@@ -701,14 +1088,164 @@ def __init__(self, scene, sim):
         self.renderer = LaserRender(self.scene.context)
         self.sim = sim
         self.matrix.post_cat(scene.context.device.device_to_scene_matrix())
+        self.last_msg = None
 
     def process_draw(self, gc: wx.GraphicsContext):
-        if self.sim.progress > 1:
+        if self.sim.progress >= 0:
+            residual = 0
+            idx = 0
             if self.sim.progress < self.sim.max:
-                sim_cut = self.sim.cutcode[: self.sim.progress - 1]
+                idx, residual = self.sim.progress_to_idx(self.sim.progress)
+                # print(f"SimWidget, idx={idx}, residual={residual:.3f}")
+                sim_cut = self.sim.cutcode[:idx]
             else:
                 sim_cut = self.sim.cutcode
             self.renderer.draw_cutcode(sim_cut, gc, 0, 0)
+            if residual > 0:
+                # We draw interpolated lines to acknowledge we are in the middle of a cut operation
+                starts = []
+                ends = []
+                cutstart = wx.Point2D(self.sim.cutcode[idx].start)
+                cutend = wx.Point2D(self.sim.cutcode[idx].end)
+                if self.sim.statistics[idx]["type"] == "RasterCut":
+                    # Rastercut object.
+                    x = 0
+                    y = 0
+                    cut = self.sim.cutcode[idx]
+                    image = cut.image
+                    gc.PushState()
+                    matrix = Matrix.scale(cut.step_x, cut.step_y)
+                    matrix.post_translate(
+                        cut.offset_x + x, cut.offset_y + y
+                    )  # Adjust image xy
+                    gc.ConcatTransform(wx.GraphicsContext.CreateMatrix(gc, ZMatrix(matrix)))
+                    try:
+                        cache = cut.cache
+                        cache_id = cut.cache_id
+                    except AttributeError:
+                        cache = None
+                        cache_id = -1
+                    if cache_id != id(image):
+                        # Cached image is invalid.
+                        cache = None
+                    if cache is None:
+                        # No valid cache. Generate.
+                        cut._cache_width, cut._cache_height = image.size
+                        try:
+                            cut.cache = LaserRender.make_thumbnail(image, maximum=5000)
```

Comment:
This should be `self.renderer.make_thumbnail` failing to do that it should crash since there's no call to `self`.